### ISSUE TITLE
lv2: notification of parameter changes to UI

### ISSUE THREAD

**Issue by jpcima** [DISTRHO/juce#6]
When a parameter is changed by reading from an input port, the UI is not notified, and is not able to update its display consequently.

If a plugin is compiled in VST variant, there isn't such a problem.
This can be verified by running LV2 and VST variants in the carla host, and manipulating the controls on the generic UI.

There has been a previous discussion on this fork, and problem of using a parameter change listener :
https://github.com/jpcima/JUCE/commit/3d9e6751d4887a0ade3b2fe599cbb3dea5fa88c1#r30518659

**Comment by jpcima** [comment]
> hmm so that function name is indeed misleading. but then I guess the UI registers a message listener too.
I would have to try and see it. if calling that triggers a change in the UI, then we should not do it.

> notification ports exist yes, but they are also an UI thing.
note that your plugin does get "setParameterValue" function called. does new juce code tries to hide this away?

I think `setParameterValue` is the obsolete call. Mine invokes `operator=` in AudioParameter(Int,Float...) by usage of parameter wrappers, which itself wraps `setParameterNotifyingHost`.

**Comment by falkTX** [comment]
so I wonder what other plugin formats in juce are doing.
I know this API is going to be removed at some point, but I have a lot of somewhat legacy plugins that make use of them.
Updating them all would be a pain.

I know camomile uses my juce lv2 fork, against latest juce, and with the new parameter API.
Might be worth a try for you.
https://github.com/pierreguillot/Camomile/tree/master/Dependencies/juce_audio_plugin_client_LV2

**Comment by jpcima** [comment]
> so I wonder what other plugin formats in juce are doing.

As I deduced from the vst wrapper some time ago, it would use parameters normalized all in floating 0-1 range. The host wouldn't know about min max and such, it would have a string conversion function hooked in the vst instance to have value displays on the host.

> I know camomile uses my juce lv2 fork, against latest juce, and with the new parameter API.
Might be worth a try for you.

Oh sure thanks, it is going to be useful. It was a long time ago that I copied the LV2 wrapper on top of a more recent Juce, for some reason I don't recall exactly, possibly some x11 related fix.
